app/Api/Controllers/WebApiController.php

```diff
@@ -34,4 +36,25 @@ public function users(Request $request): JsonResponse
             'data' => $request->input(),
         ], 501);
     }
+
+    public function login(Request $request): JsonResponse
+    {
+        $request->validate([
+            'username' => 'required|string',
+            'password' => 'required|string',
+        ]);
+
+        $credentials = [
+            Fortify::username() => $request->input('username'),
+            'password' => $request->input('password'),
+        ];
+
+        if (Auth::attempt($credentials)) {
+            $user = Auth::user();
+
+            return response()->json(['webApiKey' => $user->api_token]);
+        }
+
+        return response()->json(['error' => 'Something went wrong'], 400);
```

Comment:
I understand the necessity to not reveal too much, but would it be correct to return "Invalid user/password" here [like we do for connect](https://github.com/RetroAchievements/RAWeb/blob/master/tests/Feature/Connect/LoginTest.php#L100-L204)?

Reply:
I know @luchaos wants to move to a response structure more in line with the [official JSON API docs](https://jsonapi.org/format/#error-objects): https://discord.com/channels/476211979464343552/757767535293890682/1132370669938671758.

I think a response formatted in that manner would look like:
```
{
    "errors": [
        {
            "status": "401",
            "code": "invalid_credentials",
            "title": "Invalid username/password",
        }
    ]
}
```

I would really like to get his feedback on anything we're going to call V2 of the API.

Reply:
Connect also has a secondary "access denied" error if the credentials are valid, but the account's email address hasn't yet been verified.

Reply:
An alternative to putting this in the V2 API namespace might be to return something like `APIKey` from `r=login2`. I would be happy to put this in the V1 Web API, but it's designed to only accept unauthenticated GET requests.